linx-impulse/loaders/products/productList.ts

```diff
@@ -0,0 +1,249 @@
+import type {
+  Product,
+  ProductDetailsPage,
+  ProductListingPage,
+  PropertyValue,
+} from "../../../commerce/types.ts";
+import type { AppContext } from "../../mod.ts";
+import { getDeviceIdFromBag } from "../../utils/deviceId.ts";
+import getSource from "../../utils/source.ts";
+import { toProduct } from "../../utils/transform.ts";
+import type { PageName } from "../../utils/types/chaordic.ts";
+import { Feature, Position } from "../../utils/types/chaordic.ts";
+
+interface BaseProps {
+  /**
+   * @hide
+   */
+  page: PageName;
+  /**
+   * @title Feature
+   * @description Search for a specific feature, if not informed or found, the first one will be used
+   */
+  feature?: Feature;
+  /**
+   * @title Position
+   * @description Position of the shelf
+   */
+  position?: Position;
+  showOnlyAvailable?: boolean;
+}
+
+/**
+ * @title Home Page
+ */
+interface HomePageProps extends BaseProps {
+  /**
+   * @hide
+   */
+  page: "home";
+}
+
+/**
+ * @title Product Page
+ */
+interface ProductPageProps extends BaseProps {
+  /**
+   * @hide
+   */
+  page: "product";
+  /**
+   * @description It is recommended to create a global loader so that the loader is not called more than once
+   */
+  loader: ProductDetailsPage | null;
+  /**
+   * @ignore
+   */
+  productId?: string;
+}
+
+/**
+ * @title Category Page
+ */
+interface CategoryPageProps extends BaseProps {
+  /**
+   * @hide
+   */
+  page: "category";
+  /**
+   * @description It is recommended to create a global loader so that the loader is not called more than once
+   */
+  loader: ProductListingPage | null;
+  /**
+   * @ignore
+   */
+  categoryIds?: string[];
+}
+
+/**
+ * @title Search Page
+ */
+interface SearchPageProps extends BaseProps {
+  /**
+   * @hide
+   */
+  page: "search";
+  /**
+   * @description It is recommended to create a global loader so that the loader is not called more than once
+   */
+  loader: ProductListingPage | null;
+  /**
+   * @ignore
+   */
+  productIds?: string[];
+}
+
+const nonNullable = <T>(value: T): value is NonNullable<T> =>
+  value != null && typeof value !== "undefined";
+const normalize = (value: string) =>
+  value.toLowerCase()
+    .normalize("NFD")
+    .replace(/[\u0300-\u036f]/g, "");
+
+/**
+ * @title Page
+ */
+type Props =
+  | HomePageProps
+  | ProductPageProps
+  | CategoryPageProps
+  | SearchPageProps;
```

Comment:
Bad say that, but we already have problems doing that, can you change to:
```suggestion
type Props = { props: HomePageProps | ProductPageProps | CategoryPageProps | SearchPageProps;
```